### PR TITLE
Deploy registry backed by Cinder volume

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

**Comment by jprovaznik** [comment]
Not sure yet how far it's related to this patch, but I'm hitting following behavior with it:
after ansible finishes, registry pod is running, but the registry-deploy pod is in "DeadlineExceeded" state, the router pod (albeit it was started during ansible) is not present (was deleted)

it might be it's caused by an openshift-ansible bug - e.g. https://github.com/openshift/openshift-ansible/issues/1878

**Comment by jprovaznik** [comment]
This patch is still failing, currently on ansible run:
PLAY [Set schedulability] ****************************************************\* 

GATHERING FACTS **************************************************************\* 
<cinder-origin-master-0.example.com> ESTABLISH CONNECTION FOR USER: centos
<cinder-origin-master-0.example.com> REMOTE_MODULE setup
<cinder-origin-master-0.example.com> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o 
KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 cinder-origin-m
aster-0.example.com /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1463990181.01-188663907582731 && echo $HOME/.ansible/tmp/ansible-tmp-1463990181.01-188663907582731'
<cinder-origin-master-0.example.com> PUT /tmp/tmpmHoYNZ TO /home/centos/.ansible/tmp/ansible-tmp-1463990181.01-188663907582731/setup
<cinder-origin-master-0.example.com> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o 
KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 cinder-origin-m
aster-0.example.com /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=jnsfayxkdkcfqftowjkgddfksnxzcglg] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-jnsfayxkdkcfqftowj
kgddfksnxzcglg; LANG=C LC_CTYPE=C /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1463990181.01-188663907582731/setup; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1463990181.01-18866
3907582731/ >/dev/null 2>&1'"'"''
ok: [cinder-origin-master-0.example.com]

TASK: [Wait for master API to become available before proceeding] ************\* 
skipping: [cinder-origin-master-0.example.com]

TASK: [openshift_manage_node | Wait for Node Registration] *******************\* 
...
Result from run 49 is: {'cmd': ['oc', 'get', 'node', 'cinder-origin-node-151ply1z.example.com'], 'end': '2016-05-23 08:05:44.001095', 'stdout': u'', 'changed': True, 'attempts': 49, 'start': '2016-05-23 08:05:43.602158', 'delta': '0:00:00.398937', 'stderr': 'Error from server: nodes "cinder-origin-node-151ply1z.example.com" not found', 'rc': 1, 'warnings': []}
<cinder-origin-master-0.example.com> REMOTE_MODULE command oc get node cinder-origin-node-151ply1z.example.com
<cinder-origin-master-0.example.com> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 cinder-origin-master-0.example.com /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1463990749.76-58329254159058 && echo $HOME/.ansible/tmp/ansible-tmp-1463990749.76-58329254159058'
<cinder-origin-master-0.example.com> PUT /tmp/tmpIyMGp4 TO /home/centos/.ansible/tmp/ansible-tmp-1463990749.76-58329254159058/command
<cinder-origin-master-0.example.com> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 cinder-origin-master-0.example.com /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=bounrdvfvwatvfuapntgnoubxzrzqzdt] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-bounrdvfvwatvfuapntgnoubxzrzqzdt; LANG=C LC_CTYPE=C /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1463990749.76-58329254159058/command; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1463990749.76-58329254159058/ >/dev/null 2>&1'"'"''
Result from run 50 is: {'cmd': ['oc', 'get', 'node', 'cinder-origin-node-151ply1z.example.com'], 'end': '2016-05-23 08:05:49.649185', 'stdout': u'', 'changed': True, 'attempts': 50, 'start': '2016-05-23 08:05:49.249845', 'delta': '0:00:00.399340', 'stderr': 'Error from server: nodes "cinder-origin-node-151ply1z.example.com" not found', 'rc': 1, 'warnings': []}
failed: [cinder-origin-master-0.example.com] => (item=cinder-origin-node-151ply1z.example.com) => {"attempts": 50, "changed": false, "cmd": ["oc", "get", "node", "cinder-origin-node-151ply1z.example.com"], "delta": "0:00:00.399340", "end": "2016-05-23 08:05:49.649185", "failed": true, "item": "cinder-origin-node-151ply1z.example.com", "rc": 1, "start": "2016-05-23 08:05:49.249845", "stdout_lines": [], "warnings": []}
stderr: Error from server: nodes "cinder-origin-node-151ply1z.example.com" not found
msg: Task failed as maximum retries was encountered

FATAL: all hosts have already failed -- aborting

PLAY RECAP *******************************************************************\* 
           to retry, use: --limit @/root/main.retry

cinder-origin-master-0.example.com : ok=548  changed=126  unreachable=0    failed=1  
cinder-origin-node-151ply1z.example.com : ok=131  changed=41   unreachable=0    failed=0  
localhost                  : ok=32   changed=9    unreachable=0    failed=0

File: fragments/infra-boot.sh [hunk 1]
```diff
@@ -198,6 +198,11 @@ function clone_openshift_ansible() {
 # =============================================================================
 # MAIN
 # =============================================================================
+function sudo_enable_from_ssh() {
+    sed -i "/requiretty/s/^/#/" /etc/sudoers
+}
+
+sudo_enable_from_ssh
```
Comment (markllama) [inline fragments/infra-boot.sh:182]: Can this be in cloud-init for the infra-server? (or all?) It takes no parameters and it's universal now.

I understand cloud-init is very full for infra.

File: templates/var/lib/ansible/group_vars/OSv3.yml [hunk 2]
```diff
@@ -1,3 +1,5 @@
+ansible_first_run: {{ansible_first_run}}
```
Comment (markllama) [inline templates/var/lib/ansible/group_vars/OSv3.yml:1]: Is there a way to detect this? It may be needed but it smells bad.

Comment (lebauce) [inline templates/var/lib/ansible/group_vars/OSv3.yml:1]: I do agree, it does not look very good but I don't see a simpler way to do it

File: fragments/infra-boot.sh (hunk 1 above)
Comment (lebauce) [inline fragments/infra-boot.sh:182]: This will be removed when https://github.com/redhat-openstack/openshift-on-openstack/pull/81 is merged

**Comment by markllama** [comment]
LGTM

**Pull request merged**